hack/install-vm.sh

```diff
@@ -141,17 +150,238 @@ EOF
     systemctl status "$ANTREA_AGENT"
 }
 
+# Generic function to execute command with timeout.
+execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        echoerr "$cmd failed with status $ret at $(date), exiting"
+        exit 2
+    fi
+}
+
+# Install docker as container runtime.
+install_container_runtime() {
+    echo "Installing container runtime on $OS_NAME"
+    case "$OS_NAME" in
+        Ubuntu)
+        # Uninstall older versions of Docker(called docker, docker.io, or docker-engine).
+        docker_uninstall_packages="docker docker-engine docker.io containerd runc"
+        docker_install_packages="docker-ce docker-ce-cli containerd.io"
+        dependent_packages="ca-certificates curl gnupg lsb-release"
+        docker_gpg_key="/usr/share/keyrings/docker-archive-keyring.gpg"
+        docker_download_url="https://download.docker.com/linux/ubuntu"
+        apt-get remove ${docker_uninstall_packages}
```

Comment:
why cant we use execute_command for this?

Reply:
As we discussed if the package is not installed it throws the error and stopped the script.